src/Models/UserModel.php

```diff
@@ -195,6 +195,8 @@ public function activate(User $user): void
      * fields if they've been modified.
      *
      * @param array|User $data
+     *
+     * @TODO can't change the return type to void.
```

Comment:
I still think we should find a way to handle this with model callback events so we can catch `update()` etc. That would solve the return typing.